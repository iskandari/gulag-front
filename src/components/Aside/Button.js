import styled from 'styled-components';

export default styled.button`
  display: block;
  padding: 0;
  border: none;
  margin-bottom: 40px;

  font-size: 20px;
  font-weight: 900;
  color: #e2f3e3;

  background-color: transparent;

  opacity: 0.8;

  cursor: pointer;

  &:last-child {
    margin-bottom: 0;
  }

  &:hover {
    opacity: 0.6;
  }

  &:focus {
    outline: none;
  }
`;
