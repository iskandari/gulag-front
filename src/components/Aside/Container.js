import transition from 'styled-transition-group';

export default transition.div.attrs({
  timeout: 400
})`
  position: absolute;
  left: ${({ isMenuOpen }) => (isMenuOpen ? 0 : '-400px')};

  display: flex;
  flex-direction: column;
  justify-content: space-between;

  width: 400px;
  height: 100%;
  padding: 50px 40px;

  background-color: #14171a;
  
  &:enter {
    left: -400px;
  }

  &:enter-active {
    left: 0;

    transition: left 400ms ease-in;
  }

  &:exit {
    left: 0;
  }

  &:exit-active {
    left: -400px;

    transition: left 400ms ease-in;
  }
  
  @media (max-width: 600px) {
    width: 100%;
    left: ${({ isMenuOpen }) => (isMenuOpen ? 0 : '-100%')};
    padding-left: 10vw;
    padding-right: 10vw;
    
    &:enter {
      left: -100%;
    }

    &:enter-active {
      left: 0;
    }
    
    &:exit {
      left: 0;
    }

    &:exit-active {
      left: -100%;
    }
  }
`;
