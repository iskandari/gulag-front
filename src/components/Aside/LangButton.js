import styled from 'styled-components';

export default styled.button`
  padding: 0 0 5px;
  border: none;
  border-bottom: ${({ isActive }) => (isActive ? '5px' : 0)} solid rgba(226,243,227,0.8);
  margin: 0 25px 0 0;

  font-size: 20px;
  font-weight: 900;
  color: rgba(226,243,227,0.8);

  background-color: transparent;
`;
