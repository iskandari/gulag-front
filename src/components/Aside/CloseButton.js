import styled from 'styled-components';
import Button from '../Button';

export default styled(Button)`
  position: absolute;
  top: 40px;
  right: 30px;
  padding: 0;
  border: none;
  margin: 0;
  background-color: transparent;
  transition: 0.5s;
`;
