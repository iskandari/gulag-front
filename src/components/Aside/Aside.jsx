import React from 'react';
import PropTypes from 'prop-types';
import { t } from '../../intl/helper';

// ico
import cross from '../cross.svg';

// styled
import Container from './Container';
import CloseButton from './CloseButton';
import Button from './Button';
import LangButton from './LangButton';

const Aside = props => {
  const {
    isMenuOpen,
    openCampFilters,
    pushToSearch,
    closeMenu,
    pushToChronology,
    pushToAbout,
    locale,
    changeLang
  } = props;

  return (
    <Container isMenuOpen={isMenuOpen} in={isMenuOpen}>
      <CloseButton onClick={closeMenu}>
        <img src={cross} alt='close' />
      </CloseButton>
      <div>
        <Button onClick={pushToSearch}>{t('aside.search')}</Button>
        <Button onClick={openCampFilters}>{t('aside.campTypes')}</Button>
        <Button onClick={pushToChronology}>{t('aside.gulagChronology')}</Button>
      </div>
      <div>
        <Button onClick={pushToAbout}>{t('aboutCard.heading')}</Button>
        <div>
          <LangButton
            isActive={locale === 'ru'}
            onClick={changeLang.bind(null, 'ru')}
          >
            Rus
          </LangButton>
          <LangButton
            isActive={locale === 'en'}
            onClick={changeLang.bind(null, 'en')}
          >
            Eng
          </LangButton>
          {/* <LangButton
            isActive={locale === 'de'}
            onClick={changeLang.bind(null, 'en')}
          >
            Deu
          </LangButton> */}
        </div>
      </div>
    </Container>
  );
};

Aside.propTypes = {
  isMenuOpen: PropTypes.bool.isRequired,
  pushToSearch: PropTypes.func.isRequired,
  closeMenu: PropTypes.func.isRequired,
  openCampFilters: PropTypes.func.isRequired,
  pushToChronology: PropTypes.func.isRequired,
  pushToAbout: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired,
  changeLang: PropTypes.func.isRequired
};

export default Aside;
