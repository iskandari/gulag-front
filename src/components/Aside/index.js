import createImmutableSelector from 'create-immutable-selector';
import { push } from 'react-router-redux';
import { updateIntl } from 'react-intl-redux';
import { connect } from 'react-redux';

// selectors
import { isMenuOpenSelector } from '../App/reducers/uiReducer';
import { localeSelector } from '../App/reducers/intlReducer';

// actions
import { toggleMenu, toggleCampFilters } from '../App/reducers/uiActions';

import en from '../../intl/en';
import ru from '../../intl/ru';

import Aside from './Aside';

const locales = { en, ru };

const mapStateToProps = createImmutableSelector(
  isMenuOpenSelector,
  localeSelector,
  (isMenuOpen, locale) => ({ isMenuOpen, locale })
);
const mapDispatchToProps = dispatch => ({
  pushToSearch: () => dispatch(push('/search')),
  closeMenu: () => dispatch(toggleMenu()),
  openCampFilters: () => dispatch(toggleCampFilters()),
  pushToChronology: () => dispatch(push('/chronology')),
  pushToAbout: () => dispatch(push('/about')),
  changeLang: lang =>
    dispatch(updateIntl({ locale: lang, messages: locales[lang].messages }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Aside);
