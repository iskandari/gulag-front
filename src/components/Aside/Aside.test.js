import React from 'react';
import renderer from 'react-test-renderer';

import Aside from './Aside';

const fn = jest.fn();

test.skip('Aside renders correctly', () => {
  const component = renderer.create(
    <Aside
      isMenuOpen={false}
      pushToSearch={fn}
      closeMenu={fn}
      openCampFilters={fn}
      pushToChronology={fn}
      pushToAbout={fn}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.isMenuOpen = true;
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
