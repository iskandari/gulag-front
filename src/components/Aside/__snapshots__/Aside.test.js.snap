// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside renders correctly 1`] = `
<div
  className="sc-bdVaJa kFtmJW"
  timeout={400}
>
  <button
    className="sc-bwzfXH kSHzym"
    onClick={[Function]}
  >
    <img
      alt="close"
      src="cross.svg"
    />
  </button>
  <div>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Поиск
    </button>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Типы лагерей на карте
    </button>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Хронология ГУЛАГа
    </button>
  </div>
  <div>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      О проекте
    </button>
    <div>
      <button
        className="sc-bxivhb lkPidi"
      >
        Rus
      </button>
      <button
        className="sc-bxivhb jpemKf"
      >
        Eng
      </button>
      <button
        className="sc-bxivhb jpemKf"
      >
        Deu
      </button>
    </div>
  </div>
</div>
`;

exports[`Aside renders correctly 2`] = `
<div
  className="sc-bdVaJa kFtmJW"
  timeout={400}
>
  <button
    className="sc-bwzfXH kSHzym"
    onClick={[Function]}
  >
    <img
      alt="close"
      src="cross.svg"
    />
  </button>
  <div>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Поиск
    </button>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Типы лагерей на карте
    </button>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      Хронология ГУЛАГа
    </button>
  </div>
  <div>
    <button
      className="sc-htpNat gkmNkY"
      onClick={[Function]}
    >
      О проекте
    </button>
    <div>
      <button
        className="sc-bxivhb lkPidi"
      >
        Rus
      </button>
      <button
        className="sc-bxivhb jpemKf"
      >
        Eng
      </button>
      <button
        className="sc-bxivhb jpemKf"
      >
        Deu
      </button>
    </div>
  </div>
</div>
`;
